pkg/controller/operators/catalog/operator.go

```diff
@@ -881,6 +881,28 @@ func (o *Operator) ensureInstallPlan(logger *logrus.Entry, namespace string, sub
 	for _, installPlan := range installPlans {
 		if installPlan.Status.CSVManifestsMatch(steps) {
 			logger.Infof("found InstallPlan with matching manifests: %s", installPlan.GetName())
+
+			ownerWasAdded := false
+			for _, sub := range subs {
+				if ownerWasAdded == false && ownerutil.EnsureOwner(installPlan, sub) == false {
+					ownerWasAdded = true
+				}
+			}
+			if ownerWasAdded {
+				_, err := o.client.OperatorsV1alpha1().InstallPlans(installPlan.GetNamespace()).Update(installPlan)
+				if err != nil {
+					return nil, err
+				}
+			}
+
+			installPlan.Status.Phase = v1alpha1.InstallPlanPhaseInstalling
+			for _, step := range installPlan.Status.Plan {
+				step.Status = v1alpha1.StepStatusUnknown
+			}
+			_, err = o.client.OperatorsV1alpha1().InstallPlans(namespace).UpdateStatus(installPlan)
```

Comment:
Yes, the steps need to all be set to not completed status.

Reply:
@jpeeler could you please explain a little more:

> Yes, the steps need to all be set to not completed status.

In the case where an InstallPlan already has all of the provided Subscriptions as owners, why would we set its steps states to unknown? Are we always trying to get InstallPlans to re-apply?

Reply:
ensureInstallPlan only gets called when there's a subscription change, right? Either way, the reapplication only occurs for an existing install plan with matching manifests. Reapplying an install plan (via setting previously executed steps as unknown) is the only way to get an operator to reinstall unless the way owner references are done currently is changed, specifically with regard to other subscriptions.

Reply:
> ensureInstallPlan only gets called when there's a subscription change, right?

It gets called any time a Subscription is synced, which is on any change and also every 15m (whatever the resync interval is), right?

Reply:
Ah, that makes sense. Thanks for explaining. I'm only asking because from a user of the `ensureInstallPlan`(s) I'd expect only the InstallPlans that have changed to be re-installed, but it may not be a big deal for all of them to attempt a re-install though.

Reply:
@jpeeler I believe leaving this as you have it solves a separate bug on our backlog too, actually!

This will check/recreate any objects that are in the bundle, so if they get inadvertently deleted this will recreate them.